src/Internal/Payment/State/InitialState.php

```diff
@@ -218,6 +234,42 @@ protected function process_order_phone_number(): void {
 		}
 	}
 
+	/**
+	 * Verifies the minimum order amount, and catches the error without reaching the API.
+	 *
+	 * @throws StateTransitionException
+	 */
+	protected function verify_minimum_amount(): void {
+		$context        = $this->get_context();
+		$currency       = $context->get_currency();
+		$order_amount   = (int) $context->get_amount();
+		$minimum_amount = $this->minimum_amount_service->get_cache( $currency );
+
+		if ( $minimum_amount > $order_amount ) {
+			$error_message = $this->minimum_amount_service->get_error_message_for_shoppers( $currency, $minimum_amount );
+
+			throw new StateTransitionException( $error_message );
+		}
+	}
+
+	/**
+	 * Processes the legacy minimum amount exception from the server API.
+	 *
+	 * @param Amount_Too_Small_Exception $e Legacy Minimum Amount Exception.
+	 *
+	 * @return void
+	 * @throws StateTransitionException
+	 */
+	protected function process_minimum_amount_exception( Amount_Too_Small_Exception $e ): void {
+		$minimum_amount = $e->get_minimum_amount();
+		$currency       = $e->get_currency();
+
+		$this->minimum_amount_service->set_cache( $currency, $minimum_amount );
+		$error_message = $this->minimum_amount_service->get_error_message_for_shoppers( $currency, $minimum_amount );
+
+		throw new StateTransitionException( $error_message );
```

Comment:
We'll need to heavily look into `get_filtered_error_message` as a part of the errors stream, and it will likely be implemented somewhere else. For now, I'd be fine with simply re-throwing the original `Amount_Too_Small_Exception` after caching the amount.